relay-server/src/actors/store.rs

```diff
@@ -56,12 +55,8 @@ impl Producer {
         for topic in KafkaTopic::iter()
             .filter(|t| **t != KafkaTopic::Outcomes || **t != KafkaTopic::OutcomesBilling)
         {
-            let kafka_config = &config
-                .kafka_config(*topic)
-                .map_err(|_| ServiceError::Kafka)?;
-            client_builder = client_builder
-                .add_kafka_topic_config(*topic, kafka_config)
-                .map_err(|_| ServiceError::Kafka)?
+            let kafka_config = &config.kafka_config(*topic)?;
+            client_builder = client_builder.add_kafka_topic_config(*topic, kafka_config)?;
```

Comment:
This change helps to get a more specific error when the kafka config cannot be loaded. For example,

```
error=failed to create kafka producer: invalid kafka config
error.sources=[Client config error: No such configuration property: "foo" foo value]
```

instead of just

```
error=could not initialize kafka producer
```